mbedtls/build.rs

```diff
@@ -6,32 +6,58 @@
  * option. This file may not be copied, modified, or distributed except
  * according to those terms. */
 
+use std::collections::hash_map::DefaultHasher;
 use std::collections::{HashMap, HashSet};
-use std::env;
+use std::hash::{Hash, Hasher};
 
 use rustc_version::Channel;
+use std::env;
+use std::path::Path;
 
-/// Return the crate hash that Cargo will be passing to `rustc -C metadata=`.
-// If there's a panic in this code block, that means Cargo's way of running the
-// build script has changed, and this code should be updated to handle the new
-// case.
-fn get_compilation_metadata_hash() -> String {
+/// Retrieves or generates a metadata value used for symbol name mangling to ensure unique C symbols.
+/// When building with Cargo, the metadata value is extracted from the OUT_DIR environment variable.
+/// For Bazel builds, this method approximates Cargo's -Cmetadata by hashing the crate name and version,
+/// which are sufficient for ensuring symbol uniqueness.
+fn get_compilation_symbol_suffix() -> String {
     let out_dir: std::path::PathBuf = std::env::var_os("OUT_DIR").unwrap().into();
+    let out_dir_str = out_dir.to_string_lossy();
+
     let mut out_dir_it = out_dir.iter().rev();
-    assert_eq!(out_dir_it.next().unwrap(), "out");
-    let crate_ = out_dir_it.next().unwrap().to_string_lossy();
-    assert!(crate_.starts_with("mbedtls-"));
-    crate_[8..].to_owned()
+    let last_part = out_dir_it.next().unwrap();
+
+    if last_part == "out" {
+        // If Cargo is used as build system.
+        let crate_ = out_dir_it.next().unwrap().to_string_lossy();
+        assert!(crate_.starts_with("mbedtls-"), "Expected directory to start with 'mbedtls-'");
+        return crate_[8..].to_owned(); // Return the part after "mbedtls-"
+    } else if out_dir_str.contains("bazel-out") {
+        // If Bazel is used as build system.
+        let path_to_hash = Path::new(out_dir_str.as_ref());
+        let parts_to_hash = path_to_hash
+            .to_str()
```

Comment:
This seems unnecessary.

Reply:
Partially left from the previous method. Fixed now.